tweak/refactor: goon vampire and new vampire now has common code

dageavtobusnick opened the pull request:
<!-- Пишите **НИЖЕ** заголовков и **ВЫШЕ** комментариев, иначе ваш текст может не отобразиться. -->
<!-- В Contributing.MD вы можете найти некоторые рекомендации к оформлению пулл-реквеста. -->

## Описание
1) Гун и новый вампир теперь оба работают на коде нового вампира, перенесенного в общий класс vampire.
2) Траллы вампиров теперь так же работают на общем классе thrall.
3) Убраны ненужные проверки на гун вампира, тк теперь достаточно проверять на общий класс, в котором прописаны все взаимодействия.
4) В общем классе переведена большая часть сообщений на основе перевода, имевшегося в классе гун вампира.
5) Гун вампир теперь имеет анимацию сосания от нового вампира. 
6) Оповещение о становлении траллом, а так же сообщение о том, что игрок перестал им быть теперь находятся в датуме тралла.
<!-- Опишите, что делает ваш ПР. Документировать каждую деталь не требуется, просто укажите основные изменения. -->

## Ссылка на предложение/Причина создания ПР
Рефактор, перевод и визуальное изменение для гун вампира в виде добавления анимации от нового. Не демаю, что для этого нужна предложка.
<!-- Здесь оставьте ссылку на сообщение в #отчеты-по-предложениям, чтобы подтвердить, что ваше предложение одобрено. Либо укажите, почему этот ПР должен пройти без предложки. -->
<!-- Пример ссылки: https://discord.com/channels/617003227182792704/755125334097133628/ID-сообщения -->

dageavtobusnick committed:
refactor: goon vampire and new vampire now has general code

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into vampires_refactor

Den109G commented on code/game/objects/items/weapons/holy_weapons.dm:
```diff
@@ -581,13 +570,6 @@
 		return
 
-	for(var/mob/living/carbon/human/target in range(5, loc))
-		var/datum/antagonist/goon_vampire/g_vamp = target.mind?.has_antag_datum(/datum/antagonist/goon_vampire)
```
Это не рефактор, если механ вырезается.

dageavtobusnick replied:
Упс. Случайно видимо вырезал. Не заметил, что там нет варианта с негуновским вампиром. Сейчас верну.

dageavtobusnick replied:
Вернул

Den109G commented on code/game/objects/items/weapons/holy_weapons.dm:
```diff
@@ -580,15 +569,13 @@
-		var/datum/antagonist/goon_vampire/g_vamp = target.mind?.has_antag_datum(/datum/antagonist/goon_vampire)
-		if(g_vamp && !g_vamp.get_ability(/datum/goon_vampire_passive/full))
-			g_vamp.nullified = max(5, g_vamp.nullified + 2)
+		var/datum/antagonist/vampire/vamp = target.mind?.has_antag_datum(/datum/antagonist/vampire)
```
Нет, это не рефактор, если новых вампиров (не гуновских) задевает.

dageavtobusnick replied:
Поправил.

Den109G commented on code/datums/mind.dm:
```diff
@@ -417,7 +417,7 @@
 /datum/mind/proc/memory_edit_goon_vampire(mob/living/carbon/human/H)
 	. = _memory_edit_header("goonvampire")
-	var/datum/antagonist/goon_vampire/g_vamp = has_antag_datum(/datum/antagonist/goon_vampire)
+	var/datum/antagonist/vampire/goon_vampire/g_vamp = has_antag_datum(/datum/antagonist/vampire)
```
Он ошибочно будет думать, что обычный вамп - это гуновский.

dageavtobusnick replied:
Исправил.

Den109G commented on code/modules/antagonists/vampire/vampire_powers/goon_vampire_powers.dm:
```diff
@@ -1,75 +1,3 @@
-			to_chat(user, span_warning("Но вы же мертвы!"))
-		return FALSE
-
-	if(vampire.nullified && !fullpower)
```
У старых вампиров нельзя использовать способности если `nullified` хоть чему-то равно.
У новых - когда `nullified` достигает 100-и.

dageavtobusnick replied:
Исправил

Den109G commented on code/modules/antagonists/vampire/vampire_powers/goon_vampire_powers.dm:
```diff
@@ -1,75 +1,3 @@
-		return
-
-	var/datum/antagonist/goon_vampire/vamp = user?.mind?.has_antag_datum(/datum/antagonist/goon_vampire)
-	vamp?.bloodusable -= required_blood
```
Ну и у старых вампов не увеличивается стоимость способок.

dageavtobusnick replied:
Ну теперь у гунов использование способности не состоится, если есть хоть какая-то нулификация, а значит и уменьшения стоимости не будет.

dageavtobusnick committed:
some_fixes

Gottfrei commented on code/datums/spell_handler/vampire_spell_handler.dm:
```diff
@@ -12,41 +12,49 @@
 
 
 /datum/spell_handler/vampire/spend_spell_cost(mob/user, obj/effect/proc_holder/spell/spell)
+	for(var/datum/action/spell_action/action in user.actions)
+		action.UpdateButtonIcon()
```
зачем нам это здесь?

dageavtobusnick replied:
Было перенесено с обработчика гуновских вампиров. На всякий случай. Убрал.

Bizzonium committed:
Merge branch 'master220' into vampires_refactor

Bizzonium committed:
Merge branch 'master220' into vampires_refactor

Bizzonium committed:
Merge branch 'master220' into vampires_refactor

Gottfrei commented on code/game/objects/items/weapons/holy_weapons.dm:
```diff
@@ -49,11 +49,6 @@
 		vamp.adjust_nullification(30 + sanctify_force, 15 + sanctify_force)
 		return
 
-	var/datum/antagonist/goon_vampire/g_vamp = target.mind?.has_antag_datum(/datum/antagonist/goon_vampire)
-	if(ishuman(user) && g_vamp && !g_vamp.get_ability(/datum/goon_vampire_passive/full))
-		to_chat(target, span_warning("The nullrod's power interferes with your own!"))
-		g_vamp.nullified = max(5, g_vamp.nullified + 2)
-
```
посмотри внимательно на числа сверху и снизу.

dageavtobusnick replied:
Поправил. Теперь нулификации разные в зависимости от ее типа.

Gottfrei commented on code/modules/antagonists/goon_vampire/goon_vampire_datum.dm:
```diff
@@ -1,357 +0,0 @@
-		vamp_burn(10)
-	else		//You're in trouble, get out of the sun NOW
-		to_chat(owner.current, span_userdanger("Ваше тело обугливается, превращаясь в пепел! Укройтесь от звёздного света!"))
-		owner.current.adjustCloneLoss(10)	//I'm melting!
```
старовампиры не сгораю в космосе.

dageavtobusnick replied:
Испрвил. Теперь зависит от переменной, которая у разных вампов разная.

Gottfrei commented on code/modules/antagonists/vampire/vampire_datum.dm:
```diff
@@ -369,7 +369,9 @@
 		passive.owner = owner.current
 		passive.on_apply(src)
-
+	if(spell)
+		for(var/datum/action/spell_action/action in owner.current.actions)
+			action.UpdateButtonIcon()
```
зачем здесь это?

dageavtobusnick replied:
Было перенесено с датума гуновских вампиров. На всякий случай. Убрал.

Gottfrei commented on code/modules/antagonists/vampire/vampire_datum.dm:
```diff
@@ -414,8 +416,12 @@
 		var/level = upgrade_tiers[ptype]
 		if(bloodtotal >= level)
 			add_ability(ptype)
+			for(var/datum/action/spell_action/action in owner.current.actions)
+				action.UpdateButtonIcon()
```
зачем нам это везде где можно?

Gottfrei replied:
описание спелла и всё остальное обновляется при получении спелла, а не здесь.

dageavtobusnick replied:
Было перенесено с датума гуновских вампиров. На всякий случай. Убрал.

Gottfrei commented on code/modules/reagents/chemistry/reagents/water.dm:
```diff
@@ -377,48 +376,6 @@ GLOBAL_LIST_INIT(diseases_carrier_reagents, list(
-					M.Jitter(60 SECONDS)
-					if(prob(40))
-						M.emote("scream")
-					g_vamp.nullified = max(5, g_vamp.nullified + 2)
```
нулификация разная.

dageavtobusnick replied:
Поправил. Теперь нулификации разные. в зависимости от их типа.

Gottfrei commented on code/modules/reagents/chemistry/reagents/water.dm:
```diff
@@ -452,19 +409,6 @@ GLOBAL_LIST_INIT(diseases_carrier_reagents, list(
-				return
-			else
-				to_chat(target, "<span class='warning'>Something holy interferes with your powers!</span>")
-				g_vamp.nullified = max(5, g_vamp.nullified + 2)
```
разная

dageavtobusnick replied:
Тут она вроде как раз была +/- одинаковая. Разве что у старой не было порога. Теперь наличие верхнего порога так же зависит от типа нуллификации.

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into vampires_refactor

Gottfrei commented on code/datums/spell_handler/vampire_spell_handler.dm:
```diff
@@ -44,9 +45,14 @@
 
-	vampire.bloodusable -= calculate_blood_cost(vampire)
+
+/datum/spell_handler/vampire/revert_cast(mob/living/carbon/user, obj/effect/proc_holder/spell/spell)
+	var/datum/antagonist/vampire/vampire = user?.mind?.has_antag_datum(/datum/antagonist/vampire)
+	vampire?.bloodusable += required_blood
```
этого нет у нового вампира.

dageavtobusnick replied:
Сейчас поправлю. Есть вопросец. Если у вампира полная сила, должена ли нуллификация увеличивать стоимость способности? Просто сейчас, судя по тому, что я прочитал, стоимость увеличивается даже у тех, у кого полная сила.

Gottfrei replied:
да, должна.

dageavtobusnick replied:
Понял. Тогда уберу повышение только у старого вампира. Переопределение реверта так же перенес в обработчик способностей старого вампира.

dageavtobusnick committed:
refactor fixes

Bizzonium commented:
![image](https://github.com/user-attachments/assets/413fd088-205b-4eb8-b6a6-8a8ee20a587c)

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into vampires_refactor

dageavtobusnick commented:
> ![image](https://private-user-images.githubusercontent.com/8555356/354891228-413fd088-205b-4eb8-b6a6-8a8ee20a587c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI3NjY5NjgsIm5iZiI6MTcyMjc2NjY2OCwicGF0aCI6Ii84NTU1MzU2LzM1NDg5MTIyOC00MTNmZDA4OC0yMDViLTRlYjgtYjZhNi04YThlZTIwYTU4N2MucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MDRUMTAxNzQ4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Yzk4OGJlOGEwZWUxNDU2Nzg0MDc4MjQwOTQzYzk3MDBiMzBkNWYxNzAxYjFhZmVlYjgzY2ZlOWEyOGVlZWVhOCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.y156A7wTiDZQoXRLB555dnj7fbcm3vw-gbxlAEaRc3s)

исправлено

dageavtobusnick committed:
Merge branch 'master220' into vampires_refactor

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into vampires_refactor

dageavtobusnick committed:
fix

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into vampires_refactor

dageavtobusnick committed:
Concat vampire and goonvampire category in TP

Removing the "gun vampire" category in the trainer panel. Adding Vampire Gunas as an option for the Vampire category. Displaying the role of the vampire gunas in the "vampire" category.

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into vampires_refactor

dageavtobusnick commented:
@Bizzonium ПР уже месяц на тестмердже. Ничего из багов обнаружено не было. Может уже мердж на мейны? На полумертвых серверах уже врядли что-либо ещё найдется.
